### PR TITLE
NIFI-3502: Upgrading D3 version

### PR DESCRIPTION
**Pull request by mcgilman**
NIFI-3502:
- Upgrading to D3 version 4.

### CODE REVIEW

**Comment by scottyaslan** [comment]
Will review...

**Comment by andrewmlim** [comment]
Started to test this out.  One thing I am noticing that is different from 1.5.0 is that if I select multiple processors, then deselect everything, the operate palette still says "Multiple components selected" for a couple seconds before it eventually clears

**Comment by andrewmlim** [comment]
Upon further testing, it is really any selection and deselection where this issue occurs.  The previously selected component lingers in the Operate palette even when not selected.

**Comment by mcgilman** [comment]
Just pushed a new commit that addresses @andrewmlim findings.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js [hunk 1]
```diff
@@ -996,34 +1030,42 @@
...
+
+                                if (panning === false) {
+                                    // deselect as necessary if we are not panning
+                                    nfCanvasUtils.getSelection().classed('selected', false);
+
+                                    // update URL deep linking params
+                                    nfCanvasUtils.setURLParameters();
+
+                                    // inform Angular app values have changed
+                                    nfNgBridge.digest();
```
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js]: This is unnecessary...

Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas.js]: Which part? The .digest() was necessary to address @andrewmlim comments.

**Comment by scottyaslan** [comment]
Thanks @mcgilman this has been merged to master.

**Pull request closed without merging**